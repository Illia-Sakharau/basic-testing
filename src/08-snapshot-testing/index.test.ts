// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const testResult = generateLinkedList(['val1', 'val2']);
    expect(testResult).toStrictEqual({
      next: {
        next: {
          next: null,
          value: null,
        },
        value: 'val2',
      },
      value: 'val1',
    });
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const testResult = generateLinkedList(['val1', 'val2', 'val3']);
    expect(testResult).toMatchSnapshot();
  });
});
