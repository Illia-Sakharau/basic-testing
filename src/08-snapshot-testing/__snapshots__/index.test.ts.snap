// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "val3",
    },
    "value": "val2",
  },
  "value": "val1",
}
`;
